test/extended/oauth/requestheaders.go

```diff
@@ -504,32 +558,3 @@ func getTokenFromResponse(resp *http.Response) string {
 
 	return ""
 }
-
-func waitForAuthenticationProgressing(oc *exutil.CLI, expectedProgressing configv1.ConditionStatus) {
-	err := wait.PollImmediate(time.Second, 10*time.Minute, func() (bool, error) {
-		authn, err := oc.AdminConfigClient().ConfigV1().ClusterOperators().Get(context.Background(), "authentication", metav1.GetOptions{})
-		if err != nil {
-			e2e.Logf("Error getting authentication operator: %v", err)
-			return false, err
-		}
-
-		progressing := clusteroperatorhelpers.FindStatusCondition(authn.Status.Conditions, configv1.OperatorProgressing)
-		if progressing == nil || progressing.Status != expectedProgressing {
-			e2e.Logf("Waiting for progressing condition to be %q: %s", expectedProgressing, spew.Sdump(authn.Status.Conditions))
-			return false, nil
-		}
-
-		if expectedProgressing == configv1.ConditionFalse {
-			// make additional checks on availability and degraded status
-			if clusteroperatorhelpers.IsStatusConditionFalse(authn.Status.Conditions, configv1.OperatorAvailable) ||
-				clusteroperatorhelpers.IsStatusConditionTrue(authn.Status.Conditions, configv1.OperatorDegraded) {
```

Comment:
These changes are not going to improve the failure rate of the test in the referenced BZ. The problem is that the lib-go deployment logic sets Degraded=true when the deployment is taking too long but the Degraded inertia only sets the message.

Since the failure rate is in fact caused by a premature testing when the new config did not roll out yet because the test was also waiting for degraded=False, this does not help us.

Reply:
The hypothesis of not checking Degraded is wrong in this case.

You can merge these changes because I think they are good but you will have to remove the Bugzilla reference because the patch isn't fixing the issue.

Reply:
Yes, I could remove the reference to the BZ as it indeed, might not fix it and the changes themselves are not focused on it.

My general question would be, do you think we should add an availability check or later?

Why do you think it is wrong? If the deployment takes too long, progressing is true, right? Why do we need the additional signal of Degraded?

Reply:
> My general question would be, do you think we should add an availability check or later?
which availability check?

> Why do you think it is wrong? If the deployment takes too long, progressing is true, right?
no, progressing is actually false, degraded should be true but the inertia only causes the message to appear

Reply:
This is important to clarify and I think the root of our discussion:

Why do you think we will have a
- `Progressing=False` and `Degraded=True`, 
- but due to `inertia`, we see a `Degraded=False`?

If we can clarify the question, we could add a real `Degraded` check that looks into the `Operator` `Conditions` like it is in the post mortem.

With availability check I mean `Available=True`, as the state could be `Progressing=False`, but `Available=False` too.

Reply:
> Progressing=False and Degraded=True,

Because that's how the code works today, like I mentioned in our Slack discussions, one way to go around this would be to have both be set to true.

> If we can clarify the question, we could add a real Degraded check that looks into the Operator Conditions like it is in the post mortem.

Indeed we could.

> With availability check I mean Available=True, as the state could be Progressing=False, but Available=False too.

Since there are still workloads able to serve, Availability never changes to "false" in this case.

Reply:
Ok, thx for the discussion. There is another PR which should prevent a situation where a progressing = true operator is set to progressing = false as in combination with the inertia logic, it causes a weird limbo state.

[ref](https://github.com/openshift/library-go/pull/1380)

Reply:
Thx for your patience and code pointers that!